modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPkgDialect.java

```diff
@@ -107,9 +109,25 @@ public GeoPkgDialect(JDBCDataStore dataStore) {
         geomWriterConfig = new GeoPkgGeomWriter.Configuration();
     }
 
+    public String sqlLiteVersion;
+    public boolean supportsSubSeconds;
+
+    public void setupVersion(Connection cx) throws SQLException {
```

Comment:
Does not seem to need being public, could use a javadoc, and it's not a "setup of the version" but a way to recognize it and then decide if certain abilities are found.

I'd recommend checking the PostGIS dialect it does this a lot, uses GeoTools own Version class, has version comparison, and caches the result to avoid inspecting the version over and over.